Gems/PhysX/Code/Source/System/PhysXSystem.cpp

```diff
@@ -97,7 +99,20 @@ namespace PhysX
             m_systemConfig = *physXConfig;
         }
 
-        AzFramework::AssetCatalogEventBus::Handler::BusConnect();
+        // If the settings registry isn't available, something earlier in startup will report that failure.
+        if (auto* settingsRegistry = AZ::SettingsRegistry::Get();
+            settingsRegistry != nullptr)
+        {
+            // Automatically register the event if it's not registered, because
+            // this system is initialized before the settings registry has loaded the event list.
+            AZ::ComponentApplicationLifecycle::RegisterHandler(
+                *settingsRegistry, m_componentApplicationLifecycleHandler,
+                [this]([[maybe_unused]] AZStd::string_view path, [[maybe_unused]] AZ::SettingsRegistryInterface::Type type)
+                {
+                    InitializeMaterialLibrary();
+                },
+                "LegacySystemInterfaceCreated"); // LegacySystemInterfaceCreated is signaled after critical assets have been processed
```

Comment:
Won't this just break again once CrySystem goes away? Can we use something else?

Reply:
Hopefully that notification won't just get removed without considering the systems that listen to it and placing some other alternative solution :)
At the moment I took a look and there was no other global notification after Asset Processor has processed critical assets, this is the closest notification.
I don't have enough knowledge myself of the O3DE initialization chain, but hopefully a better solution will come out of this bug: https://github.com/o3de/o3de/issues/6093

Reply:
How about adding a Lifecycle event for when critical assets shouldn't have been processed?

You can add a ComponentApplicationLifecycle event to the bottom of [CCryEditApp::CompileCriticalAssets](https://github.com/o3de/o3de/blob/development/Code/Editor/CryEdit.cpp#L1332-L1358)

```
AZ::ComponentApplicationLifecycle::SignalEvent(*settingsRegistry, "AssetProcessorIsReady", R"({})");
```

Reply:
"Hopefully that notification won't just get removed without considering the systems that listen to it and placing some other alternative solution :)" True, but ideally we shouldn't add new dependencies on legacy things we know are going away, it's just kicking the can down the road:)

But if there is no other sensible place for this to happen at the moment and it isn't simple to add a new notification at the appropriate place then fair enough.

Reply:
For a new lifecycle event to be useful for PhysX System initialization it'd need to be triggered by both Editor and Launcher and it has to be in a place that guarantees criticals to be processed, that's why `LegacySystemInterfaceCreated` works, because it's an event after AP has been dealt with. The event would have to be called even when there is no asset processor involved, so they are triggered in release builds as well.

Separately, I put a breakpoint in `CCryEditApp::CompileCriticalAssets` and it doesn't get called when running the editor, for what I've seen the logic to wait for the AP is done in `AssetSystemComponent.cpp` at function `AssetSystemComponent::EstablishAssetProcessorConnection`.